users/tests.py

```diff
@@ -1,3 +1,85 @@
+        with pytest.raises(ValidationError):
+            new_profile.save()
+            new_profile.city = LONG_PROFESSION
+            new_profile.clean_fields()
+
+    @pytest.mark.parametrize("user_idx, account_type", [(1, Account_type.STUDENT)])
+    def test_create_new_profile(self, make_profile, user_idx, account_type):
+        profile = make_profile(user_idx, account_type)
+        assert profile.account_type == account_type
+        assert profile.user.username == f"username{user_idx}"
+        assert profile.user.first_name == f"first_name{user_idx}"
+        assert profile.user.last_name == f"last_name{user_idx}"
+        assert profile.user.email == f"email{user_idx}"
+        assert profile.city == f"city{user_idx}"
+
+    @pytest.mark.parametrize("user_idx, account_type, second_user_idx, second_account_type",
+                             [(1, Account_type.STUDENT, 2, Account_type.TEACHER)])
+    def test_filter_by_city(self, make_profile, user_idx, account_type, second_user_idx, second_account_type):
+        new_profile = make_profile(user_idx, account_type)
+        make_profile(second_user_idx, second_account_type)
+        profiles_by_city = Profile.filter_by_city(new_profile.city)
+        assert list(profiles_by_city) == [new_profile]
+
+    @pytest.mark.parametrize("user_idx, account_type, second_user_idx, second_account_type",
+                             [(1, Account_type.STUDENT, 2, Account_type.TEACHER)])
+    def test_filter_by_first_name(self, make_profile, user_idx, account_type, second_user_idx, second_account_type):
+        new_profile = make_profile(user_idx, account_type)
+        make_profile(second_user_idx, second_account_type)
+        profiles_by_first_name = Profile.filter_by_first_name(new_profile.user.first_name)
+        assert list(profiles_by_first_name) == [new_profile]
+
+    @pytest.mark.parametrize("user_idx, account_type, second_user_idx, second_account_type",
+                             [(1, Account_type.STUDENT, 2, Account_type.TEACHER)])
+    def test_filter_by_last_name(self, make_profile, user_idx, account_type, second_user_idx, second_account_type):
+        new_profile = make_profile(user_idx, account_type)
+        make_profile(second_user_idx, second_account_type)
+        profiles_by_last_name = Profile.filter_by_last_name(new_profile.user.last_name)
+        assert list(profiles_by_last_name) == [new_profile]
+
+    @pytest.mark.parametrize("user_idx, account_type, second_user_idx, second_account_type",
+                             [(1, Account_type.STUDENT, 2, Account_type.TEACHER)])
+    def test_filter_by_account_type(self, make_profile, user_idx, account_type, second_user_idx, second_account_type):
+        new_profile = make_profile(user_idx, account_type)
+        make_profile(second_user_idx, second_account_type)
+        profiles_by_account_type = Profile.filter_by_account_type(new_profile.account_type)
+        assert list(profiles_by_account_type) == [new_profile]
```

Comment:
I recommend using `sorted(<list object>)`, otherwise tests will end up flaky...

Reply:
If it’s about the order maybe a set will work too?

Reply:
It is ok to use lists with sorted. IMO sets will be confusing here.